Tests/SystemTests/PublisherAuthenticationSystemTests.swift

```diff
@@ -0,0 +1,173 @@
+        let keyTokens = Secrets.ablyApiKey.split(separator: ":")
+        let keyName = String(keyTokens[0])
+        
+        let fetchedTokenDetails = AuthHelper().requestToken(
+            options: RestHelper.clientOptions(true, key: Secrets.ablyApiKey),
+            clientId: keyName
+        )
+        
+        let connectionConfiguration = ConnectionConfiguration(clientId: keyName, authCallback: { tokenParams, resultHandler in
+            guard let tokenDetails = fetchedTokenDetails else {
+                XCTFail("TokenDetails doesn't exist")
+                return
+            }
+            
+            resultHandler(.success(.tokenDetails(tokenDetails)))
+        })
+        
+        testPublisherTrack(configuration: connectionConfiguration)
+    }
+    
+    func testPublisherConnectsWithTokenString() throws {
+        let keyTokens = Secrets.ablyApiKey.split(separator: ":")
+        let keyName = String(keyTokens[0])
+        
+        let fetchedTokenString = AuthHelper().requestToken(
+            options: RestHelper.clientOptions(true, key: Secrets.ablyApiKey),
+            clientId: keyName
+        )?.token
+                
+        let connectionConfiguration = ConnectionConfiguration(clientId: keyName, authCallback: { tokenParams, resultHandler in
+            guard let tokenString = fetchedTokenString else {
+                XCTFail("TokenDetails doesn't exist")
+                return
+            }
+            
+            resultHandler(.success(.jwt(tokenString)))
+        })
+        
+        testPublisherTrack(configuration: connectionConfiguration)
+    }
+    
+    func testPublisherConnectsWithJWT() throws {
+        guard let jwtToken = JWTHelper().getToken() else {
+            XCTFail("Create JWT failed")
+            return
+        }
+        
+        let connectionConfiguration = ConnectionConfiguration { tokenParams, resultHandler in
+            resultHandler(.success(.jwt(jwtToken)))
+        }
+        
+        testPublisherTrack(configuration: connectionConfiguration)
+    }
+    
+    private func testPublisherTrack(configuration: ConnectionConfiguration) {
+        let resolution = Resolution(accuracy: .balanced, desiredInterval: 5000, minimumDisplacement: 100)
+        let publisher = try! PublisherFactory.publishers()
+            .connection(configuration)
+            .mapboxConfiguration(MapboxConfiguration(mapboxKey: Secrets.mapboxAccessToken))
+            .log(logConfiguration)
+            .locationSource(LocationSource(locationSource: [CLLocation(latitude: 0.0, longitude: 0.0)]))
+            .routingProfile(.driving)
+            .delegate(publisherDelegate)
+            .resolutionPolicyFactory(DefaultResolutionPolicyFactory(defaultResolution: resolution))
+            .start() // Doesn't start publishing, its just a `build()` publisher call.
+
+        // TODO check that connection is made/ Await successfully connection callback with an expectation
```

Comment:
What do you think about this issue @lukasz-szyszkowski ?

Reply:
I've talked to @KacperKluka some time ago and there is no reason to check if the publisher connects successfully or not, all events are enqueued and will execute asap. But for the testing purposes, it'll be helpful, I agree.